### PR TITLE
Todo edit feature

### PR DESCRIPTION
**Pull request by valeshgopal**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Added 'edit task' feature to the 'To-do App'. Users can correct any typos or add extra info with the help of this functionality.

Fixes # (issue)
Previously there was no option to edit a task.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Click on the 'edit' icon
- Make changes to the task
- Submit the task
- Task is updated

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Programming-School-Pro-Coding — COMMENTED** [review]
Hi Sir

I suggest to use LocalStorage to save the data to the browser

If you want to more explanation tell me

**Comment by atapas** [comment]
@valeshgopal Thanks for the play. Could you please create an issue for this play?

**Review by Programming-School-Pro-Coding — DISMISSED** [review]
LGTM

**Comment by valeshgopal** [comment]
> @valeshgopal Thanks for the play. Could you please create an issue for this play?

Hi Tapas! I've opened the issue.

File: src/plays/react-todo-app/ReactTodoApp.jsx [hunk 1]
```diff
@@ -11,50 +11,81 @@ import './react-todo.css';
 import Input from './components/Input';
 import TodoList from './components/TodoList';
 
+function getLocalStorageTodos() {
```
Comment (atapas) [inline src/plays/react-todo-app/ReactTodoApp.jsx:14]: I see an existing hook to handle localstorage operation: https://github.com/reactplay/react-play/blob/main/src/plays/keeper/Hooks/useLocalStorage.js

But this hook is unfortunately inside a play. Could you please move the hook here `https://github.com/reactplay/react-play/tree/main/src/common/hooks` and then use it?

It is better we use reusable code across. Please modify the import of the existing play where it is being used once you move it.

File: src/plays/react-todo-app/ReactTodoApp.jsx [hunk 2]
```diff
@@ -11,50 +11,81 @@ import './react-todo.css';
...
+  const todos = localStorage.getItem('todos');
```
Comment (atapas) [inline src/plays/react-todo-app/ReactTodoApp.jsx:15]: Always use keys that are unique. May be another play have same key name and override yours.

**Review by atapas — CHANGES_REQUESTED** [review]
@valeshgopal

I've requested a couple of refactoring to make the code organization better. Could you please take a look.

Also the UI breaks in the responsive mode, and need a fix. If you could check this as well, will be great.

image

**Comment by valeshgopal** [comment]
Hi Tapas! I've made the following changes:
- Changed UI look to make it feel more aesthetic 
- Made the form responsive
- Moved local Storage hook from https://github.com/reactplay/react-play/blob/main/src/plays/keeper/Hooks/useLocalStorage.js to https://github.com/reactplay/react-play/tree/main/src/common/hooks and imported it from there.
- Added folder name before localstorage key to make it more unique - (eg: useLocalStorage('react-todo-app/todos', []), useLocalStorage('keeper/notes', []))

File: src/plays/react-todo-app/ReactTodoApp.jsx (hunk 2 above)
Comment (valeshgopal) [reply src/plays/react-todo-app/ReactTodoApp.jsx]: > Always use keys that are unique. May be another play have same key name and override yours.

Added folder name before localstorage key to make it more unique - (eg: useLocalStorage('react-todo-app/todos', []), useLocalStorage('keeper/notes', []))

File: src/plays/keeper/Keeper.jsx [hunk 3]
```diff
@@ -1,23 +1,23 @@
 import { getPlayById } from 'meta/play-meta-util';
 import PlayHeader from 'common/playlists/PlayHeader';
-import './Keeper.css'
+import './Keeper.css';
 import Header from './Header';
 import CreateArea from './CreateArea';
 import Note from './Notes';
-import useLocalStorage from './Hooks/useLocalStorage';
+import useLocalStorage from '../../common/hooks/useLocalStorage';
```
Comment (atapas) [inline src/plays/keeper/Keeper.jsx:7]: The import should work like this too?

import useLocalStorage from 'common/hooks/useLocalStorage';

**Comment by atapas** [comment]
> Hi Tapas! I've made the following changes:
> 
> * Changed UI look to make it feel more aesthetic
> * Made the form responsive
> * Moved local Storage hook from https://github.com/reactplay/react-play/blob/main/src/plays/keeper/Hooks/useLocalStorage.js to https://github.com/reactplay/react-play/tree/main/src/common/hooks and imported it from there.
> * Added folder name before localstorage key to make it more unique - (eg: useLocalStorage('react-todo-app/todos', []), useLocalStorage('keeper/notes', []))

I loved that. Thank You!

Given a tiny comment. Now I will test the changes from UI too.

File: src/plays/keeper/Keeper.jsx (hunk 3 above)
Comment (valeshgopal) [reply src/plays/keeper/Keeper.jsx:7]: I tried that path at first. But it gives the following error message: 
Module not found: Error: You attempted to import /common/hooks/useLocalStorage which falls outside of the project src/ directory. Relative imports outside of src/ are not supported.
You can either move it inside src/, or add a symlink to it from project's node_modules/.

Comment (atapas) [reply src/plays/keeper/Keeper.jsx:7]: Interesting: See this usage. We are doing already

https://github.com/reactplay/react-play/blob/34dbcb1b6d8035b8813e92dfdcce6f5b83215cca/src/plays/react-todo-app/ReactTodoApp.jsx#L3

Comment (atapas) [reply src/plays/keeper/Keeper.jsx:7]: Should you add hooks here?

https://github.com/reactplay/react-play/blob/main/src/common/index.js

**Comment by atapas** [comment]
@all-contributors please add @valeshgopal for Code

**Pull request merged**